plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BaseDeployPreferencesPanel.java

```diff
@@ -123,15 +123,15 @@
   private final ProjectRepository projectRepository;
   private final FormToolkit formToolkit;
 
-  public CommonDeployPreferencesPanel(Composite parent, IProject project,
+  public BaseDeployPreferencesPanel(Composite parent, IProject project,
```

Comment:
noticing that requireValues here really isn't very OO. This would normally be done as a separate subclass with different behavior.